rpcs3/Emu/RSX/Overlays/overlay_animation.h

```diff
@@ -23,30 +23,6 @@ namespace rsx
             this->z = other.z;
         }
 
-        vector3_base<T> operator * (const vector3_base<T>& rhs) const
```

Comment:
Why move this? I'm genuinely curious, I'm not very upto date on latest C++ spec recommendations so I wonder if there is a good reason for it.

Reply:
The operator below was moved so that you can do vector * T (no way to do that with an in-class definition), this one was just for consistency so that all of them are outside. Should I move out only the ones that need it?

Reply:
What do you mean vector * T? You mean like vector3f * float? That works just fine with in-class definition. In fact, all the methods you defined should work just fine with in-class unless you have a multiple typename operation which is both undesirable and not present; i.e vector3f * int for example where T=float and U=int.

Reply:
Actually now that I think about it, it would still work. I cannot think of a situation where in-class definition wouldn't work.

Reply:
Sorry, posted the comment hastily, I meant `T * vector` (in this case `float * vector3f`)
which is not possible with in-class definitions (see  https://stackoverflow.com/a/34106613/3548976).

Reply:
Fair enough, although that sounds like an artificial problem. The only situation I can imagine this being a problem is a situation where a * b is not equal to b * a which I have only encountered with some matrix transforms. That being said, this is now not a simple quick wrapper which mandates it gets moved to a utils file. Overlays/overlay_utils should be fine, you may move the vertex class from the overlay_controls file in there as well while you're at it.